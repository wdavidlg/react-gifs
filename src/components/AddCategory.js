import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = (props) => {


    const [value, setValue] = useState('');

    const handleChange = (e) => {
        setValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('handleSubmit', value);
        if(value.trim().length !== 0){
            props.setCategories((prev) => [value, ...prev]);
            setValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
           <input 
                type="text"
                placeholder="Add Category"
                value={value}
                onChange={handleChange}
           />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
