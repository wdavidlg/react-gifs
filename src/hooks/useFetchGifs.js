import  {useEffect, useState} from 'react';
import {getGifs} from '../helpers/getGifs'


export const useFetchGifs = (category) => {

    const [state, setState] = useState({data: [], loading: true});
    useEffect( () => {
        getGifs(category).then( gifs => {
            setTimeout( () => {
                //console.log(gifs)
                setState({
                    data: gifs,
                    loading: false
                });
            }, 2000)
        })
    }, [category] );
        
    return state;
}