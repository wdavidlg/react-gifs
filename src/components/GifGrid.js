import React from 'react'
import Gif from './Gif'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
    

    const {data:gifs, loading} = useFetchGifs(category);

    const getCardGifs = () => {
        return gifs.map((item) => <Gif key={item.id} gif={item} />);
    }
    console.log(loading, category);
    return (
        <div className='grid'>
            <h3 className='animate__animated animate__fadeInBottomRight'>{category}</h3>
            {loading && <p className='animate__animated animate__flash animate__delay-1s'>Cargando...</p>}
            <div className='category'>{getCardGifs()}</div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


