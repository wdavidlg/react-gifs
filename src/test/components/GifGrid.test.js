import React from 'react';
import {shallow} from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas con <GifGRid />', () => {
    
    const category = 'Naruto';

    test('debe de mostrar el componente correctamente', () => {
        //Tiene que ir antes de la definicion del wrapper
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        }) //Esta llamada es la de por defecto
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan los gifs', () => {
       
        const gifs = [{
            id: 'ABC',
            url: 'https://imagen.jpg',
            title: 'Mi titulo'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category} />)

        console.log(wrapper.find('div').length)
        expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('p').exists()).toBe(false);
       expect(wrapper.find('Gif').length).toBe(gifs.length);

    })
    
    
})
