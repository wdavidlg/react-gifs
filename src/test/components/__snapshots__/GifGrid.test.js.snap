// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con <GifGRid /> debe de mostrar el componente correctamente 1`] = `
<div
  className="grid"
>
  <h3
    className="animate__animated animate__fadeInBottomRight"
  >
    Naruto
  </h3>
  <p
    className="animate__animated animate__flash animate__delay-1s"
  >
    Cargando...
  </p>
  <div
    className="category"
  />
</div>
`;

exports[`pruebas con <GifGRid /> debe de mostrar items cuando se cargan los gifs 1`] = `
<div
  className="grid"
>
  <h3
    className="animate__animated animate__fadeInBottomRight"
  >
    Naruto
  </h3>
  <div
    className="category"
  >
    <Gif
      gif={
        Object {
          "id": "ABC",
          "title": "Mi titulo",
          "url": "https://imagen.jpg",
        }
      }
      key="ABC"
    />
  </div>
</div>
`;
